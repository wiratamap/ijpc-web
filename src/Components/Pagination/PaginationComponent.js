import React from 'react';
import PropTypes from 'prop-types';

const PaginationComponent = ({
  previousButtonDisabled, nextButtonDisabled, onClickPreviousButton, onClickNextButton
}) => (
  <nav aria-label="Pagination">
    <ul className="pagination pagination-lg justify-content-end">
      <li className="page-item">
        <button
          type="button"
          className="page-link"
          aria-label="Previous"
          disabled={previousButtonDisabled}
          onClick={onClickPreviousButton}
        >
          <span aria-hidden="true">&laquo;</span>
        </button>
      </li>
      <li className="page-item">
        <button
          type="button"
          className="page-link"
          aria-label="Next"
          disabled={nextButtonDisabled}
          onClick={onClickNextButton}
        >
          <span aria-hidden="true">&raquo;</span>
        </button>
      </li>
    </ul>
  </nav>
);

PaginationComponent.propTypes = {
  previousButtonDisabled: PropTypes.bool,
  nextButtonDisabled: PropTypes.bool,
  onClickPreviousButton: PropTypes.func,
  onClickNextButton: PropTypes.func
};

PaginationComponent.defaultProps = {
  previousButtonDisabled: true,
  nextButtonDisabled: true,
  onClickPreviousButton: Function,
  onClickNextButton: Function
};

export default PaginationComponent;
