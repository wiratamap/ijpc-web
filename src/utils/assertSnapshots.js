import React from 'react';
import renderer from 'react-test-renderer';

/**
 * @param Component - (Component) Component that needs to be test.
 * @param testConfiguration - (object) object of props and test description
 */
const assertSnapshot = (Component, testConfiguration) => {
  const { props, description } = testConfiguration;

  describe('#render', () => {
    it(description, () => {
      const component = renderer.create(<Component {...props} />).toJSON();

      expect(component).toMatchSnapshot();
    });
  });
};

const assertSnapshots = (Component, testConfigurations) => {
  testConfigurations.forEach((testConfiguration) => assertSnapshot(Component, testConfiguration));
};

export default assertSnapshots;
