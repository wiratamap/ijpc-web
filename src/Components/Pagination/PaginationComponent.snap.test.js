import assertSnapshots from '../../utils/assertSnapshots';
import PaginationComponent from './PaginationComponent';

describe('PaginationComponent', () => {
  const configurations = [
    {
      props: {
        previousButtonDisabled: true,
        nextButtonDisabled: true,
        onClickPreviousButton: jest.fn(),
        onClickNextButton: jest.fn()
      },
      description: 'should render pagination component with defined props'
    }
  ];

  assertSnapshots(PaginationComponent, configurations);
});
