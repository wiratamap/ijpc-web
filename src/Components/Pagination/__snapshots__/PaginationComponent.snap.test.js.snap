// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationComponent #render should render pagination component with defined props 1`] = `
<nav
  aria-label="Pagination"
>
  <ul
    className="pagination pagination-lg justify-content-end"
  >
    <li
      className="page-item"
    >
      <button
        aria-label="Previous"
        className="page-link"
        disabled={true}
        onClick={[MockFunction]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        aria-label="Next"
        className="page-link"
        disabled={true}
        onClick={[MockFunction]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
      </button>
    </li>
  </ul>
</nav>
`;
